bundles/tools.vitruv.change.correspondence/src/tools/vitruv/change/correspondence/model/PersistableCorrespondenceModelImpl.java

```diff
@@ -0,0 +1,181 @@
+		if (resourceUri != null) {
+			this.correspondencesResource = withGlobalFactories(new ResourceSetImpl()).createResource(resourceUri);
+			this.correspondencesResource.getContents().add(correspondences);
+		} else {
+			this.correspondencesResource = null;
+		}
+	}
+
+	@Override
+	public void loadSerializedCorrespondences(ResourceSet resolveIn) {
+		checkState(correspondencesResource != null, "Correspondences resource must be specified to load existing correspondences");
+		Resource loadedResource = loadOrCreateResource(withGlobalFactories(new ResourceSetImpl()), correspondencesResource.getURI());
+		if (!loadedResource.getContents().isEmpty()) {
+			Correspondences loadedCorrespondences = (Correspondences) loadedResource.getContents().get(0);
+			for (Correspondence correspondence : loadedCorrespondences.getCorrespondences()) {
+				List<EObject> resolvedLeftObjects = resolve(correspondence.getLeftEObjects(), resolveIn);
+				replace(correspondence.getLeftEObjects(), resolvedLeftObjects);
+				List<EObject> resolvedRightObjects = resolve(correspondence.getRightEObjects(), resolveIn);
+				replace(correspondence.getRightEObjects(), resolvedRightObjects);
+			}
+			this.correspondences.getCorrespondences().addAll(loadedCorrespondences.getCorrespondences());
+		}
+	}
+
+	private static void replace(List<EObject> originalList, List<EObject> elementsToReplaceWith) {
+		originalList.clear();
+		originalList.addAll(elementsToReplaceWith);
+	}
+
+	private static List<EObject> resolve(List<EObject> eObjects, ResourceSet resolveIn) {
+		List<EObject> resolvedEObjects = eObjects.stream().map(eObject -> EcoreUtil.resolve(eObject, resolveIn)).collect(Collectors.toList());
+		for (EObject resolvedEObject : resolvedEObjects) {
+			checkState(!resolvedEObject.eIsProxy(), "object %s is referenced in correspondence but could not be resolved", resolvedEObject);
+		}
+		return resolvedEObjects;
+	}
+
+	@Override
+	public void save() {
+		removeCorrespondencesForRemovedElements();
+		if (this.correspondencesResource != null) {
+			try {
+				this.correspondencesResource.save(null);
+			} catch (IOException e) {
+			}
+		}
+	}
+
+	@Override
+	public <C extends Correspondence> C addCorrespondenceBetween(List<EObject> firstEObjects, List<EObject> secondEObjects, String tag,
+			Supplier<C> correspondenceCreator) {
+		C correspondence = correspondenceCreator.get();
+		correspondence.getLeftEObjects().addAll(firstEObjects);
+		correspondence.getRightEObjects().addAll(secondEObjects);
+		correspondence.setTag(tag);
+		this.correspondences.getCorrespondences().add(correspondence);
+		return correspondence;
+	}
+
+	private void removeCorrespondencesForRemovedElements() {
+		Iterator<Correspondence> iterator = correspondences.getCorrespondences().iterator();
+		while (iterator.hasNext()) {
+			Correspondence element = iterator.next();
+			if (element.getLeftEObjects().stream().anyMatch(this::isNotInManagedResource)
```

Comment:
Actually, it is not regardless of the result. The check works as follows: We check whether all elements in a correspondence are still in a managed resource, then, of course, everything is fine. If some of the elements are not in a managed resource anymore (i.e., when some of the elements have been removed), the all elements must have been removed to automatically remove the correspondence, as otherwise we are in some inconsistent state (some of the elements of a correspondence have been removed, but the correspondence is still there).